new/analyse/add-dependencies.ts

```diff
@@ -1,64 +1,73 @@
-import { Module, Package, RelatedPackageTypes } from '../types';
+import {
+  Module,
+  Package,
+  RelatedPackageDependsOn,
+  RelatedPackageTypes,
+} from '../types';
 
 import { logger } from '../helpers/logger';
 
-const { error, log } = logger('add dependencies');
+const { error, log } = logger('[analyse] add dependencies');
 
 export const addDependencies: Module = (env) => {
+  log("Populating packages' dependencies");
+
   if (!env.packages) {
     const errorMessage = '"packages" attribute not found on env config object';
     error(errorMessage);
     throw new Error(errorMessage);
   }
 
-  log("Populating packages' dependencies");
-
   const packagesNames = Object.keys(env.packages);
 
-  const packages = packagesNames.reduce(
-    (accumulator, packageName, _currentIndex, packagesNames) => {
-      const currentPackage = { ...(accumulator![packageName] as Package) };
-
-      // Assign only internal dependencies
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.dependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'default' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+  const packages = packagesNames.reduce((accumulator, packageName) => {
+    const currentPackage = { ...(accumulator![packageName] as Package) };
 
-      // Assign only internal devDependencies
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.devDependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'dev' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+    const dependencies: RelatedPackageDependsOn[] = [
+      // all dependencies
+      ...Object.entries(currentPackage.packageJson.dependencies || {}).map(
+        ([name, range]) => ({
```

Comment:
Are external dependencies also needed/wanted here?

Reply:
They're filtered out below for both types at the same time.